Feature/component data for 3rd party components (#44)
AdaloHQ/docs: website/docs/api-reference/configuration/manifest-json.md

| Key                 | Type | Description                                                                                                                                                                                                                                             |
| ------------------- | ---- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `listItem`          |      | This prop will be a list child: See [Lists]()                                                                                                                                                                                                           |

> | `formValue`         |      |Instead of the normal value, this prop will pass a `{ value, onChange }` object that can be used like a react [Controlled Component](). `value` will initially be given the value passed to the prop and can have its value exported throughout the app. `onChange` is the event handler function that can be used to update `value`. |                                                                                    

Reviewer: ```suggestion
| `formValue`         |      | Instead of the normal value, this prop will pass a `{ value, onChange }` object that can be used like a react [Controlled Component](). `value` will initially be given the value passed to the prop and can have its value exported throughout the app. `onChange` is the event handler function that can be used to update `value`. |                                                                                    
```

> | `formValue`         |      | Instead of the normal value, this prop will pass a `{ value, onChange, initial }` object that can be used like a react [Controlled Component](). `value` will initially be given the value passed to the prop and can have its value exported throughout the app, `onChange` is the event handler function that can be used to update `value`, and `initial` is the initial value passed to the prop that will not change when `onChange` is called. |                                                                                    

Reviewer 2: ```suggestion
| `formValue`         |      | Instead of just the value itself, this prop will pass a `{ value, onChange, initial }` object that can be used like a React [Controlled Component](https://reactjs.org/docs/forms.html#controlled-components). `value` will initially be given the value passed to the prop and that value can be exported throughout the app. `onChange` is the event handler function that can be used to update `value`. `initial` is the initial value passed to the prop that will not change when `onChange` is called. |                                                                                    
```

Might wanna link the Controlled Component to react docs here? @scott-massey what do you think?

Reviewer 2: Should the `formValue` now come after `autosaveInput` since the `formValue` has one more thing (`initialValue`) than `autosaveInput`?

> | `autosaveInput`     |      | Instead of the normal value, this prop will pass a `{ value, onChange }` object that can be used like a react [Controlled Component](). Value will be typed based on `type`, and `onChange` will be a function that takes a single argument, `newValue` |

Reviewer 2: ```suggestion
| `autosaveInput`     |      | Instead of just the value itself, this prop will pass a `{ value, onChange }` object that can be used like a react [Controlled Component](). Value will be typed based on `type`, and `onChange` will be a function that takes a single argument, `newValue` |
```

const { testAction } = this.props;
testAction(arg1, arg2);
```

## Component Data

Components that intend to export data throughout an app must implement the `formValue` prop in order to do so.

### `formValue` Prop

> The value that a form component intends to export must be given the role of `formValue`. This value will be accessible to other components in the form of Magic Text. Instead of the normal value, component itself will be passed a `{ value, onChange }` object that can be used like a react [Controlled Component](), as well as `${propName}_initial`. `value` will store the data that the component exports, it will be set to the props value by default and be updated by `onChange`, which should be called as an event handler function. `${propName}_initial` is a static value that will always be equal to the initial value of `value`.

Reviewer 2: ```suggestion
The value that a form component intends to export must be given a `role` of `formValue`. This value will be accessible to other components in the form of Magic Text. Instead of the normal value, the component itself will be passed a `{ value, onChange }` object that can be used like a react [Controlled Component](), as well as `${propName}_initial`. `value` will store the data that the component exports, it will be set to the props value by default and be updated by `onChange`, which should be called as an event handler function. `${propName}_initial` is a static value that will always be equal to the initial value of `value`.
```

For convenience / repetition's sake, you could also just link `formValue` stuff here or just copy over most of the same wording from above so it's consistent!
